nanosim_models/lsff_interventions.py

```diff
@@ -0,0 +1,379 @@
+# iron_conc_distributions = {
+#     'India': stats.uniform(loc=14, scale=21.5-14), # Uniform(14,21.5) mg iron as NaFeEDTA per kg flour
+#     'Ethiopia': stats.bernoulli(p=0,loc=30), # 30 mg iron as NaFeEDTA per kg flour,
+#     'Nigeria': stats.bernoulli(p=0,loc=40), # 40 mg iron as NaFeEDTA per kg flour,
+# }
+
+def get_iron_concentration(location, draws):
+    """
+    Get the iron concentration in flour for specified location (mg iron as NaFeEDTA per kg flour),
+    with parameter uncertainty if there is any.
+    
+    Returns
+    -------
+    scalar if there is no uncertainty, or pandas Series indexed by draw if there is uncertainty.
+    """
+    if location == 'India':
+        iron_conc_dist = stats.uniform(loc=14, scale=21.5-14) # Uniform(14,21.5) mg iron as NaFeEDTA per kg flour
+#         if take_mean: # we'd have to pass another argument
+#         if isinstance(draws, pd.CategoricalIndex): # We used a categorical index if we took the mean over draws
+        if len(draws) == 1: # Use our best guess if there's only one draw or we took the mean
+            iron_concentration = iron_conc_dist.mean()
+        else:
+            iron_concentration = pd.Series(
+                iron_conc_dist.rvs(size=len(draws)), index=draws, name='iron_concentration')
+    elif location == 'Ethiopia':
+        iron_concentration = 30 # 30 mg iron as NaFeEDTA per kg flour
+    elif location == 'Nigeria':
+        iron_concentration = 40 # 40 mg iron as NaFeEDTA per kg flour
+    else:
+        raise ValueError(f'Unsupported location: {location}')
+
+    return iron_concentration
+
+def sample_flour_consumption(location, sample_size):
+    """Sample from distribution of daily flour consumption (in Ethiopia).
+    The distribution is uuniform between each quartile: min=0, q1=77.5, q2=100, q3=200, max=350.5
+    This distribution represents individual heterogeneity and currently has no parameter uncertainty.
+
+    Currently, the data is hardcoded, but eventually it should be location-dependent.
+    The Ethiopia data comes from the Ethiopian National Food Consumption Survey (2013).
+    """
+#     # TODO: Edit this to simply call the propensity version below
+#     # Define quartiles in g of flour per day
+#     q = (0, 77.5, 100, 200, 350.5) # min=0, q1=77.5, q2=100, q3=200, max=350.5
+#     u = np.random.uniform(0,1,size=sample_size)
+#     # Scale the uniform random number to the correct interval based on its quartile
+#     return np.select(
+#         [u<0.25, u<0.5, u<0.75, u<1],
+#         [q[1]*u/0.25, q[1]+(q[2]-q[1])*(u-0.25)/0.25, q[2]+(q[3]-q[2])*(u-0.5)/0.25, q[3]+(q[4]-q[3])*(u-0.75)/0.25]
+#     )
+    return get_flour_consumption_from_propensity(location, np.random.uniform(0,1,size=sample_size))
+
+def get_flour_consumption_from_propensity(mean_consumption, propensity):
+    """Get distribution of daily flour consumption (in Ethiopia) based on the specified propensity array.
```

Comment:
Consider typing arguments (https://docs.python.org/3/library/typing.html)
It seems weird to me that on line 85 `get_flour_consumption_from_propensity` is called with "location" but on line 87 the same argument is "mean consumption"

Reply:
Ah, good catch. This is a result of refactoring my code -- in the old version the argument was "location" (but was really just a placeholder that was unused), and in the new version I changed it to "mean_consumption". In the update, I ended up not using that function that still calls the "location" version, so I never bothered updating that function. When I have a chance I'll clean up the file to be more sensible.

And yes, type hints would be a good idea. Since I'm not as familiar with the language as I would be coming from the engineering side, I've found that looking up the right way to do it (syntax, type classes, etc.) has been too much overhead to maintain as I'm iteratively changing function signatures and trying to get everything to work... Maybe once I have some time to invest in thoroughly doing it once, it'll become easier to maintain type hints as I go...